### PR TITLE
Add pemutil.UnbundleCertificate

### PR DESCRIPTION
**Pull request by areed**
Removes a certificate from a bundle.

#### Name of feature:
pemutil.UnbundleCertificate.

#### Pain or issue this feature alleviates:
BundleCertificate lets you create a new bundle with additional certificates but there was no way to remove certificates from a bundle.

#### Supporting links/other PRs/issues:

💔Thank you!

### CODE REVIEW

File: pemutil/pem.go [hunk 1]
```diff
@@ -740,3 +740,38 @@ func BundleCertificate(bundlePEM []byte, certsPEM ...[]byte) ([]byte, bool, erro
...
+		sum := sha256.Sum224(cert.Raw)
+		if drop[sum] {
+			modified = true
+			continue
+		}
+		block, err := Serialize(cert)
+		if err != nil {
+			return nil, false, err
+		}
+		keep = append(keep, pem.EncodeToMemory(block)...)
```
Comment (maraino) [inline pemutil/pem.go:773]: Simplify these lines with:
```suggestion
		keep = append(keep, pem.EncodeToMemory(&pem.Block{
			Type: "CERTIFICATE", 
			Bytes: cert.Raw,
		})...)
```

File: pemutil/pem.go [hunk 2]
```diff
@@ -740,3 +740,38 @@ func BundleCertificate(bundlePEM []byte, certsPEM ...[]byte) ([]byte, bool, erro
 
 	return bundlePEM, modified, nil
 }
+
+// UnbundleCertificate removes PEM-encoded certificates from a PEM-encoded
+// certificate bundle.
+func UnbundleCertificate(bundlePEM []byte, certsPEM ...[]byte) ([]byte, bool, error) {
+	drop := make(map[[sha256.Size224]byte]bool, len(certsPEM))
+	for i := range certsPEM {
+		cert, err := ParseCertificate(certsPEM[i])
```
Comment (maraino) [inline pemutil/pem.go:749]: I can see using this function but having `certsPEM[i]` as a certificate bundle. If a read a bundle and want to remove one cert, I don't want to split the bundle in multiple certs.

File: pemutil/pem.go [hunk 3]
```diff
@@ -740,3 +740,38 @@ func BundleCertificate(bundlePEM []byte, certsPEM ...[]byte) ([]byte, bool, erro
 
 	return bundlePEM, modified, nil
 }
+
+// UnbundleCertificate removes PEM-encoded certificates from a PEM-encoded
+// certificate bundle.
+func UnbundleCertificate(bundlePEM []byte, certsPEM ...[]byte) ([]byte, bool, error) {
```
Comment (maraino) [inline pemutil/pem.go:746]: Optionally, return the original bundle if `len(certsPEM) == 0`

**Review by maraino — CHANGES_REQUESTED** [review]

File: pemutil/pem.go (hunk 1 above)
Comment (areed) [reply pemutil/pem.go]: 8b54772

File: pemutil/pem.go (hunk 3 above)
Comment (areed) [reply pemutil/pem.go:746]: 8b54772

File: pemutil/pem.go (hunk 2 above)
Comment (areed) [reply pemutil/pem.go]: 9286a0c

**Pull request merged**